katdal/visdatav4.py

```diff
@@ -399,6 +418,37 @@ def _before(date):
         # on selection in the process
         self.select(spw=0, subarray=0, ants=obs_ants)
 
+    def _register_standard_cal_streams(self):
+        freqs = self.spectral_windows[0].channel_freqs
+        attrs = self.source.metadata.attrs
+        archived_streams = attrs.get('sdp_archived_streams', [])
+        # The default L1 cal stream, useful for older files
+        l1_stream = 'cal'
```

Comment:
What will happen with files that don't have any cal stream (e.g. because there are <4 antennas, or Mattieu has turned it off for holography)?

Reply:
A 'cal' view will be established on telstate but it will be empty, prompting `add_applycal_sensors` to skip the virtual sensors and return `None`.

I improved it so that the `l1` stream won't be registered as an entry in `cal_freqs` either, which will help point out the missing stream when parsing user input.